server/common.js

```diff
@@ -137,11 +157,42 @@ var isEmptyObject = function(obj) {
     }
     return true;
 }
-
 exports.isEmptyObject = isEmptyObject;
 
 // return the current date
 function getDate() {
     return date().format('YYYY-MM-DD hh:mm:ss A');
 }
 exports.getDate = getDate;
+// </Global Function> -----------------------------------------------
+
+// <File System functions> ------------------------------------------
+// make a directory given the path and the name of the new directory
+var mkdir = function (parentPath, directoryName, callback) {
+    var fullPath = path.join(parentPath, directoryName);
```

Comment:
are these variables given by the user? 
Don't want any shell hijacking happening here. If it is, please validate them before executing any code.

Reply:
good point. how can we check this? This function is backend only, but the other function do use some user input

Reply:
Check by seeing if the code you will execute has any special characters in it. It should only be alphabet strings. Disregard any Pipes, slashes, periods, etc...